sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -1008,6 +1008,7 @@ def _upload_blob_chunk(self, location: str, data: IO[bytes], **kwargs) -> Tuple[
                 location,
                 BytesIO(buffer),
                 cls=_return_response_headers,
+                retry_on_methods=["PATCH"],
```

Comment:
Is there a reason to add it? It does not help users  (ACR does not really support retries on unload yet)  and customizes core policy

Reply:
Do we need to validate if `retry_on_methods` in `kwargs`?

Reply:
> Is there a reason to add it? It does not help users (ACR does not really support retries on upload yet) and customizes core policy

We need it so that it will retry when an error occurs and can get the 404 error.

Reply:
the 404 error is a consequence of service not supporting retries on upload, there is no need to match this behavior seem on other languages. Instead we can document on python that uploads are not retried and should be restarted from scratch

Reply:
> Do we need to validate if `retry_on_methods` in `kwargs`?

`retry_con_methods` is consumed from `kwargs` in [`azure-core`](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/azure/core/pipeline/policies/_retry.py#L100). I've checked it did passed to `azure-core` when debugging. Does it make sense?

Reply:
If there is a `retry_con_methods` in `kwargs`, this call will raise. I think the question is do we allow users to specify `retry_con_methods` in the operation?